gtm/variable_template.tpl

```diff
@@ -318,12 +330,41 @@ scenarios:
       assertThat(resp).isString();
       assertThat(resp).isEqualTo("2");
     });
+- name: Test case if 'addRequestDataToItems' is false to show Vertex AI behaves as
```

Comment:
Thanks for doing this but the other tests also need updating. I've rewritten the test logic to help out. Let me share what I've done here. Would you mind updating it.

Setup:
```
const JSON = require("JSON");
const Promise = require("Promise");

const purchasedProducts = [];
let vertexAIResonse;
let expectedPayload;

/**
 * Build the mock items from the event.
 * This method changes the global purchasedProducts variable, which is used in
 * the mock logic.
 * @param {!Array<number>} items - the items to mock.
 */
function generateMockItems(items) {
  for (const item of items) {
    purchasedProducts.push({
      "item_id": item.item_id,
      "item_name": item.item_name,
      "price": item.revenue,
      "quantity": item.quantity
    });
  }
}

/**
 * Build the mock response from VertexAI.
 * This method changes the global vertexAIResonse variable, which is used in
 * the mock logic.
 * @param {!Array<number>} predictions - the predictions to use in the response.
 * @param {number} statusCode - the status code to use in the VertexAI response.
 */
function generateMockVertexAI(predictions, statusCode) {
  const predictionString = predictions.join(', ');
  vertexAIResonse = {
    "statusCode": statusCode,
    "body":"{\"predictions\": [" + predictionString + "]}"
  };
}

// Change sendHttpRequest to return our mocked VertexAI response.
mock("sendHttpRequest", (url, requestOptions, postBody) => {
  assertThat(postBody).isEqualTo(expectedPayload);
  return Promise.create((resolve) => {
    resolve(vertexAIResonse);
  });
});

// Inject our products into the event data.
mock("getEventData", (data) => {
  if (data === "items") {
    return purchasedProducts;
  }
});
```

Test: `Prediction data is added to items`
```
const mockVariableData = {
  projectNumber: "11111111111",
  vertexEndpointID: "1234567891011121314",
  cloudLocation: "europe-west2",
  addRequestDataToItems: true,
  data: [
    {"key": "browser", "value": "Chrome"},
  ]
};

generateMockItems([
  {"item_id": "shoes", "item_name": "Shoes", "revenue": 80, "quantity": 2},
]);

generateMockVertexAI([80], 200);

expectedPayload = JSON.stringify({
  "instances":[{
    "item_id": "shoes",
    "item_name": "Shoes",
    "price": 80,
    "quantity": 2,
    "browser": "Chrome",
    "index": 1
  }],
  "parameters": {}
});

runCode(mockVariableData).then((resp) => {
  assertThat(resp).isString();
  assertThat(resp).isEqualTo("80");
});

```

Test: `Prediction data is not added to the items`
```
const mockVariableData = {
  projectNumber: "11111111111",
  vertexEndpointID: "1234567891011121314",
  cloudLocation: "europe-west2",
  addRequestDataToItems: false,
  data: [
    {"key": "age", "value": "20"},
    {"key": "address", "value": "A"},
    {"key": "job", "value": "it"}
  ]
};

generateMockVertexAI([100], 200);

expectedPayload = JSON.stringify({
  "instances":[{
    "age": "20",
    "address": "A",
    "job": "it"
  }],
  "parameters":{}
});

runCode(mockVariableData).then((resp) => {
  assertThat(resp).isString();
  assertThat(resp).isEqualTo("100");
});

```

Test: `Prediction responses are summed for multiple items`
```
const mockVariableData = {
  projectNumber: "11111111111",
  vertexEndpointID: "1234567891011121314",
  cloudLocation: "europe-west2",
  addRequestDataToItems: true
};

generateMockItems([
  {"item_id": "shoes", "item_name": "Shoes", "revenue": 80, "quantity": 2},
  {"item_id": "blazer", "item_name": "Blazer", "revenue": 100, "quantity": 1},
]);

generateMockVertexAI([80, 100], 200);

expectedPayload = JSON.stringify({
  "instances":[{
    "item_id": "shoes",
    "item_name": "Shoes",
    "price": 80,
    "quantity": 2,
    "index": 1
  },
  {
    "item_id": "blazer",
    "item_name": "Blazer",
    "price": 100,
    "quantity": 1,
    "index": 2
  }],
  "parameters": {}
});

runCode(mockVariableData).then((resp) => {
  assertThat(resp).isString();
  assertThat(resp).isEqualTo("180");
});

```

Test: `Check default returned on error status code`
```
const mockVariableData = {
  projectNumber: "11111111111",
  vertexEndpointID: "1234567891011121314",
  cloudLocation: "europe-west2",
  addRequestDataToItems: true,
  defaultValueOnError: "0"
};

generateMockItems([
  {"item_id": "shoes", "item_name": "Shoes", "revenue": 80, "quantity": 2},
]);

generateMockVertexAI([], 500);

expectedPayload = JSON.stringify({
  "instances":[{
    "item_id": "shoes",
    "item_name": "Shoes",
    "price": 80,
    "quantity": 2,
    "index": 1
  }],
  "parameters": {}
});

runCode(mockVariableData).then((resp) => {
  assertThat(resp).isString();
  assertThat(resp).isEqualTo("0");
});

```